packages/checkout-storefront/src/sections/PaymentSection/PaymentMethods.tsx

```diff
@@ -1,29 +1,86 @@
 import { MessageKey, useFormattedMessages } from "@/checkout-storefront/hooks/useFormattedMessages";
-import React from "react";
-import { camelCase } from "lodash-es";
-import { PaymentMethodID } from "checkout-common";
+import React, { SyntheticEvent, useEffect, useMemo } from "react";
+import { camelCase, compact } from "lodash-es";
 import { SelectBoxGroup } from "@/checkout-storefront/components/SelectBoxGroup";
 import { SelectBox } from "@/checkout-storefront/components/SelectBox";
 import { Text } from "@saleor/ui-kit";
-import { UsePaymentMethods } from "./usePaymentMethods";
+import {
+  ChannelActivePaymentProvidersByChannel,
+  PaymentMethodID,
+  PaymentProviderID,
+} from "checkout-common";
+import { useAppConfig } from "@/checkout-storefront/providers/AppConfigProvider";
+import { useCheckout, useFetch } from "@/checkout-storefront/hooks";
+import { getPaymentMethods } from "@/checkout-storefront/fetch";
+import { AvailablePaymentMethods } from "@/checkout-storefront/sections/PaymentSection/types";
+import { CheckoutFormData } from "@/checkout-storefront/sections/CheckoutForm/types";
 
-export type PaymentMethodsProps = UsePaymentMethods;
+export interface PaymentMethodsProps {
+  selectedPaymentMethod: PaymentMethodID;
+  setValue: (key: keyof CheckoutFormData, value: PaymentProviderID | PaymentMethodID) => void;
+  onSelect: (event: SyntheticEvent) => void;
+}
 
 export const PaymentMethods: React.FC<PaymentMethodsProps> = ({
-  availablePaymentMethods,
   selectedPaymentMethod,
-  setSelectedPaymentMethod,
+  onSelect,
+  setValue,
 }) => {
   const formatMessage = useFormattedMessages();
+  const {
+    checkout: {
+      channel: { id: channelId },
+    },
+  } = useCheckout();
+
+  const {
+    env: { checkoutApiUrl },
+  } = useAppConfig();
+
+  const [{ data: allPaymentOptions, loading }] = useFetch(getPaymentMethods, {
+    args: { channelId, checkoutApiUrl },
+    skip: !channelId,
+  });
+
+  const getParsedPaymentMethods = (
+    allPaymentMethods: ChannelActivePaymentProvidersByChannel | null | undefined
+  ): AvailablePaymentMethods => {
+    if (!allPaymentMethods) {
+      return [];
+    }
+
+    return compact(Object.keys(allPaymentMethods)) as AvailablePaymentMethods;
+  };
+
+  const availablePaymentMethods = getParsedPaymentMethods(allPaymentOptions);
+
+  useEffect(() => {
+    if (!loading && allPaymentOptions && !availablePaymentMethods.length) {
+      throw new Error("No available payment providers");
+    } else if (!loading && availablePaymentMethods.length && !selectedPaymentMethod) {
+      setValue("paymentMethodId", availablePaymentMethods[0] as PaymentMethodID);
+    }
```

Comment:
I think this would make it a little bit more readable
```suggestion
    if (loading) return;
    
    if (!availablePaymentMethods.length) {
        throw new Error("No available payment providers");
    } else if (!selectedPaymetMethod) {
        setValue("paymentMethodId", availablePaymentMethods[0] as PaymentMethodID);
    }
```